base/reduce.jl

```diff
@@ -13,20 +13,23 @@ const SmallUnsigned = Union{UInt8,UInt16,UInt32}
 end
 
 const CommonReduceResult = Union{UInt64,UInt128,Int64,Int128,Float32,Float64}
-const WidenReduceResult = Union{SmallSigned, SmallUnsigned, Float16}
+const SmallReduceResult = Union{SmallSigned, SmallUnsigned, Float16}
```

Comment:
`SmallReduceResult` seems like a misnomer here.  The result is `Int`, which is *not* small.

`Float16` should still be widened to `Float32`, presumably, which is not a "system size".